showthread.php

```diff
@@ -734,15 +734,22 @@
 	}
 
 	// Increment the thread view.
-	if($mybb->settings['delayedthreadviews'] == 1)
+	if(
+		($mybb->settings['threadviews_countspiders'] == 1 && $session->is_spider == true) ||
+		($mybb->settings['threadviews_countguests'] == 1 && $mybb->user['uid'] == 0) ||
+		($mybb->settings['threadviews_countthreadauthor'] == 1 && $mybb->user['uid'] == $thread['uid'])
+	)
```

Comment:
```suggestion
	if(
		(
			$mybb->user['uid'] == 0 &&
			(
				($session->is_spider == true && $mybb->settings['threadviews_countspiders'] == 1) ||
				($session->is_spider == false && $mybb->settings['threadviews_countguests'] == 1)
			)
		) ||
		(
			$mybb->user['uid'] != 0 &&
			($mybb->settings['threadviews_countthreadauthor'] == 1 || $mybb->user['uid'] != $thread['uid'])
		)
	)
```

Addressed:
- missed visitors from none of the groups from settings
- confusing guests and spiders (`uid == 0`)
- confusing guest thread authors with other guests